src/main/java/racingcar/Car.java

```diff
@@ -1,19 +1,31 @@
 package racingcar;
 
 public class Car {
-
+    private String name;
     private int location;
 
-    public Car() {
-        this.location = 1;
+    public Car(String name) throws Exception {
+        checkCarNameLength(name);
+        this.name = name;
+        this.location = 0;
+    }
+
+    private void checkCarNameLength(String name) throws Exception {
+        if (name.length() > 5) {
+            throw new Exception("자동차 이름 길이 초과 (최대 5자)");
+        }
+    }
+
+    public String getName() {
+        return name;
     }
 
     public int getLocation() {
         return location;
     }
 
-    public void move() {
-        ++location;
+    public void move(int distance) {
+        location += distance;
     }
```

Comment:
https://github.com/next-step/java-racingcar/pull/5489#discussion_r1536543641 이전 코멘트에서 드렸던 의견은 자동차가 얼마만큼 이동할지의 값이 아니라, 자동차가 이동할지 말지를 결정할 임의의 숫자를 전달해보시란 의미였습니다. 지금은 자동차의 이동에 대한 주도권을 `CarRacing`의 `moveForward()` 메서드가 가지고 있는데, `moveForward(0`에서는 단순히 임의의 숫자 값을 생성해서 이를 자동차에게 넘겨주고 자동차가 이 숫자를 가지고 이동할지 말지를 스스로 결정핟록 해보면 좋을 것 같습니다.
```suggestion
    public void move(int randomNumber) {
        if (randomNumber > 4) {
            location++;
        }
    }
```